pkg/codegen/dotnet/templates.go

```diff
@@ -85,6 +86,17 @@ namespace {{.Namespace}}
             };
         }
 
+        public static InvokeOptions WithDefaults(this InvokeOptions? options)
+        {
+            return new InvokeOptions
+            {
+                Parent = options?.Parent,
+                Provider = options?.Provider,
+                Version = options?.Version != null ? options?.Version : Version,
```

Comment:
Nit: Could use the [null-coalescing operator](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-coalescing-operator) `??`:

```suggestion
                Version = options?.Version ?? Version,
```